#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation 1 0 0 4.71238898038469
  position -4.864362322232888 13.891122324035562 -3.0494343473603776
}
TexturedBackground {
  texture "factory"
  skybox FALSE
}
TexturedBackgroundLight {
  texture "factory"
  castShadows FALSE
}
RectangleArena {
  floorSize 15 10
  floorAppearance DEF FLOOR_COLOR PBRAppearance {
    baseColor 0.403922 0.4 0.27451
    roughness 1
    metalness 0
  }
  wallThickness 0.25
  wallHeight 2
  wallAppearance DEF WALL_COLOR PBRAppearance {
    baseColor 0.886275 0.878431 0.6
    roughness 1
    metalness 0
  }
}
RectangleArena {
  translation 2.5 0 0
  name "rectangle arena(1)"
  floorSize 10 5
  floorAppearance USE FLOOR_COLOR
  wallThickness 0.25
  wallHeight 2
  wallAppearance USE WALL_COLOR
}
RectangleArena {
  translation -6.25 0 0
  name "rectangle arena(2)"
  floorSize 2.5 5
  floorAppearance USE FLOOR_COLOR
  wallThickness 0.25
  wallHeight 2
  wallAppearance USE WALL_COLOR
}
Robot {
  translation -3.78 0.05 -2.95
  rotation 0 1 0 3.14159
  children [
    DEF PLATFORM Shape {
      appearance PBRAppearance {
        baseColor 0.886275 0.823529 0.886275
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.4 0.005 0.4
      }
    }
    DEF BODY Solid {
      translation 2.57125e-07 0.160006 -0.049997
      rotation 0 0 1 1.5708
      children [
        DEF STICK1 Solid {
          translation 0.328 -0.155 0.147
          rotation -0.016949705017968317 0.9970414716451949 -0.07497340408030169 -1.38388
          children [
            DEF STICK1_SHAPE Shape {
              appearance DEF STICK_COLOR PBRAppearance {
                baseColor 0.443137 0.431373 0.447059
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.01 0.01 1
              }
            }
          ]
          name "stick1"
          boundingObject USE STICK1_SHAPE
        }
        DEF STICK2 Solid {
          translation 0.327 0.154 0.148
          rotation 0.015039718542295452 0.9970551531988499 0.07519859271147726 -1.38
          children [
            DEF STICK2_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 1
              }
            }
          ]
          name "stick2"
          boundingObject USE STICK2_SHAPE
        }
        DEF STICK3 Solid {
          translation 0.195 -0.163 -0.02
          rotation -0.05014370195119564 0.9982887500369301 0.030086221170717384 1.23664
          children [
            DEF STICK3_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.75
              }
            }
          ]
          name "stick3"
          boundingObject USE STICK3_SHAPE
        }
        DEF STICK4 Solid {
          translation 0.192 0.1646 -0.017
          rotation 0.06992636973228512 0.9975019914833652 -0.010003989343473154 1.24
          children [
            DEF STICK4_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.75
              }
            }
          ]
          name "stick4"
          boundingObject USE STICK4_SHAPE
        }
        DEF STICK5 Solid {
          translation 0.814 5.68543e-06 0.0513
          rotation 1 0 0 1.5708
          children [
            DEF STICK5_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.23
              }
            }
          ]
          name "stick5"
          boundingObject USE STICK5_SHAPE
        }
        DEF STICK6 Solid {
          translation 0.663 5.50178e-06 0.083
          rotation 1 0 0 1.5708
          children [
            DEF STICK6_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.25
              }
            }
          ]
          name "stick6"
          boundingObject USE STICK6_SHAPE
        }
        DEF STICK7 Solid {
          translation 0.05 -1.73472e-18 0.203
          children [
            DEF STICK7_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.35 0.01
              }
            }
          ]
          name "stick7"
          boundingObject USE STICK7_SHAPE
        }
        DEF STICK8 Solid {
          translation 0.05 -0.177 0.067
          children [
            DEF STICK8_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.277
              }
            }
          ]
          name "stick8"
          boundingObject USE STICK8_SHAPE
        }
        DEF STICK9 Solid {
          translation 0.05 0.176 0.068
          children [
            DEF STICK9_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.275
              }
            }
          ]
          name "stick9"
          boundingObject USE STICK9_SHAPE
        }
        DEF STICK10 Solid {
          translation 0.05 5.93452e-06 -0.071
          rotation 1 0 0 -1.5707953071795862
          children [
            DEF STICK10_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.36
              }
            }
          ]
          name "stick10"
          boundingObject USE STICK10_SHAPE
        }
        DEF STICK11 Solid {
          translation 0.814 5.75683e-06 -0.0187
          rotation 1 0 0 -1.5707953071795862
          children [
            DEF STICK11_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.23
              }
            }
          ]
          name "stick11"
          boundingObject USE STICK11_SHAPE
        }
        DEF STICK12 Solid {
          translation 0.814 -0.1124 0.018
          rotation 1 0 0 0
          children [
            DEF STICK12_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.07
              }
            }
          ]
          name "stick12"
          boundingObject USE STICK12_SHAPE
        }
        DEF STICK13 Solid {
          translation 0.814 0.1124 0.018
          rotation 1 0 0 -3.141592653589793
          children [
            DEF STICK13_SHAPE Shape {
              appearance USE STICK_COLOR
              geometry Box {
                size 0.01 0.01 0.07
              }
            }
          ]
          name "stick13"
          boundingObject USE STICK13_SHAPE
        }
      ]
      name "body"
    }
    DEF LEFT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.19 0.0200158 0.120003
      }
      device [
        RotationalMotor {
          name "left_wheel_motor"
        }
        PositionSensor {
          name "left_wheel_position_sensor"
        }
      ]
      endPoint Solid {
        translation 0.21 0.020020500000000042 0.1200039999999998
        rotation 1 0 0 0
        children [
          DEF RUNNING_WHEEL Transform {
            rotation 0 0 1 -1.5707996938995747
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.196078 0.215686 0.215686
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.054
                  subdivision 24
                }
              }
            ]
            rotationStep 0
          }
        ]
        name "left_wheel"
        boundingObject USE RUNNING_WHEEL
        physics DEF WHEEL_PHYSICS Physics {
        }
      }
    }
    DEF RIGHT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.19 0.0200163 0.120003
      }
      device [
        RotationalMotor {
          name "right_wheel_motor"
        }
        PositionSensor {
          name "right_wheel_position_sensor"
        }
      ]
      endPoint Solid {
        translation -0.21 0.020021300000000047 0.12000399999999976
        rotation 1 0 0 0
        children [
          USE RUNNING_WHEEL
        ]
        name "RIGHT_WHEEL"
        boundingObject USE RUNNING_WHEEL
        physics USE WHEEL_PHYSICS
      }
    }
    DEF BACK_WHEEL_JOINT Hinge2Joint {
      jointParameters HingeJointParameters {
        anchor 0.01 -0.02 -0.189997
      }
      endPoint Solid {
        translation 0.01 -0.02 -0.189997
        children [
          DEF BACK_WHEEL Transform {
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.196078 0.215686 0.215686
                  roughness 1
                  metalness 0
                }
                geometry Sphere {
                  radius 0.017
                  subdivision 25
                  ico FALSE
                }
              }
            ]
          }
        ]
        name "BACK_WHEEL"
        boundingObject USE BACK_WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_FC DistanceSensor {
      translation -5.38735e-07 0.21 0.16
      rotation 0 1 0 -1.5707996938995747
      children [
        DEF DS_SENSOR_SHAPE Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.02 0.04
          }
        }
      ]
      name "ds_fc"
      lookupTable [
        0 0 0
        2 255 0
      ]
    }
    DEF DS_FL DistanceSensor {
      translation 0.099999 0.21 0.16
      rotation 0 1 0 -1.5707996938995747
      children [
        USE DS_SENSOR_SHAPE
      ]
      name "ds_fl"
      lookupTable [
        0 0 0
        2 255 0
      ]
    }
    DEF DS_FR DistanceSensor {
      translation -0.1 0.21 0.16
      rotation 0 1 0 -1.5707996938995747
      children [
        USE DS_SENSOR_SHAPE
      ]
      name "ds_fr"
      lookupTable [
        0 0 0
        2 255 0
      ]
    }
    DEF DS_LF DistanceSensor {
      translation 0.183 0.21 0.11
      children [
        USE DS_SENSOR_SHAPE
      ]
      name "ds_lf"
      lookupTable [
        0 0 0
        2 255 0
      ]
    }
    DEF DS_LB DistanceSensor {
      translation 0.183 0.21 -0.07
      children [
        USE DS_SENSOR_SHAPE
      ]
      name "ds_lb"
      lookupTable [
        0 0 0
        2 255 0
      ]
    }
    DEF DS_RF DistanceSensor {
      translation -0.181 0.21 0.11
      rotation 0 1 0 3.141592653589793
      children [
        USE DS_SENSOR_SHAPE
      ]
      name "ds_rf"
      lookupTable [
        0 0 0
        2 255 0
      ]
    }
    DEF DS_RB DistanceSensor {
      translation -0.181 0.21 -0.07
      rotation 0 1 0 -3.141592653589793
      children [
        USE DS_SENSOR_SHAPE
      ]
      name "ds_rb"
      lookupTable [
        0 0 0
        2 255 0
      ]
    }
    Receiver {
    }
  ]
  name "elc-bot"
  boundingObject USE PLATFORM
  physics Physics {
    mass 1
  }
  controller "ELCController"
}
DEF DRIVER Robot {
  children [
    Emitter {
    }
  ]
  name "Driver"
  controller "driver"
  supervisor TRUE
}
